sdk/search/search/review/search.api.md

```diff
@@ -246,11 +817,113 @@ export type SuggestResult<T> = {
+
+// @public
+export interface TokenFilter {
+    name: string;
+    odatatype: "TokenFilter";
+}
+
+// @public
+export type TokenFilterName = 'arabic_normalization' | 'apostrophe' | 'asciifolding' | 'cjk_bigram' | 'cjk_width' | 'classic' | 'common_grams' | 'edgeNGram_v2' | 'elision' | 'german_normalization' | 'hindi_normalization' | 'indic_normalization' | 'keyword_repeat' | 'kstem' | 'length' | 'limit' | 'lowercase' | 'nGram_v2' | 'persian_normalization' | 'phonetic' | 'porter_stem' | 'reverse' | 'scandinavian_normalization' | 'scandinavian_folding' | 'shingle' | 'snowball' | 'sorani_normalization' | 'stemmer' | 'stopwords' | 'trim' | 'truncate' | 'unique' | 'uppercase' | 'word_delimiter';
+
+// @public
+export type TokenFilterUnion = TokenFilter | AsciiFoldingTokenFilter | CjkBigramTokenFilter | CommonGramTokenFilter | DictionaryDecompounderTokenFilter | EdgeNGramTokenFilter | EdgeNGramTokenFilterV2 | ElisionTokenFilter | KeepTokenFilter | KeywordMarkerTokenFilter | LengthTokenFilter | LimitTokenFilter | NGramTokenFilter | NGramTokenFilterV2 | PatternCaptureTokenFilter | PatternReplaceTokenFilter | PhoneticTokenFilter | ShingleTokenFilter | SnowballTokenFilter | StemmerTokenFilter | StemmerOverrideTokenFilter | StopwordsTokenFilter | SynonymTokenFilter | TruncateTokenFilter | UniqueTokenFilter | WordDelimiterTokenFilter;
+
+// @public
+export interface TokenInfo {
+    readonly endOffset: number;
+    readonly position: number;
+    readonly startOffset: number;
+    readonly token: string;
+}
+
+// @public
+export interface Tokenizer {
+    name: string;
+    odatatype: "Tokenizer";
+}
+
+// @public
+export type TokenizerName = 'classic' | 'edgeNGram' | 'keyword_v2' | 'letter' | 'lowercase' | 'microsoft_language_tokenizer' | 'microsoft_language_stemming_tokenizer' | 'nGram' | 'path_hierarchy_v2' | 'pattern' | 'standard_v2' | 'uax_url_email' | 'whitespace';
+
+// @public
+export type TokenizerUnion = Tokenizer | ClassicTokenizer | EdgeNGramTokenizer | KeywordTokenizer | KeywordTokenizerV2 | MicrosoftLanguageTokenizer | MicrosoftLanguageStemmingTokenizer | NGramTokenizer | PathHierarchyTokenizerV2 | PatternTokenizer | StandardTokenizer | StandardTokenizerV2 | UaxUrlEmailTokenizer;
+
+// @public
+export interface TruncateTokenFilter {
+    length?: number;
+    name: string;
+    odatatype: "#Microsoft.Azure.Search.TruncateTokenFilter";
+}
+
+// @public
+export interface UaxUrlEmailTokenizer {
+    maxTokenLength?: number;
+    name: string;
+    odatatype: "#Microsoft.Azure.Search.UaxUrlEmailTokenizer";
+}
+
+// @public
+export interface UniqueTokenFilter {
+    name: string;
+    odatatype: "#Microsoft.Azure.Search.UniqueTokenFilter";
+    onlyOnSamePosition?: boolean;
+}
+
 // @public
 export interface UploadDocumentsOptions extends IndexDocuments {
     mergeIfExists?: boolean;
 }
 
+// @public
+export interface WordDelimiterTokenFilter {
+    catenateAll?: boolean;
```

Comment:
First I've heard of this variant form of "concatenate" (and now the unix util `cat` makes sense)! Though, `concat` being a thing in JS, I wonder if we should align to that?

Reply:
Given that these models are intentionally meant to mirror their Lucene counterparts, I'd be hesitant to deviate too far from the Lucene parameter names.

Reply:
Yeah, in this case I think bringing it closer to JS is going to muddle things since Indexes are very 1:1 with Lucene concepts